### ISSUE TITLE
Editor always assumes align: full is 100% width

### ISSUE THREAD

**Issue by chrisvanpatten** [WordPress/gutenberg#7374]
In Gutenberg, if a block is set to `align: full`, the editor applies CSS in other areas that assumes the "full" width will always be 100%. If your `align: full` is less, the editor display is wonky.

This causes issues for themes that might want the "full" value to not be fully edge-to-edge in the browser. (In our case, we are aligning full width to the width of a page container element.)

**To Reproduce**
Set your `align: full` width a specific value less than 100%:

```css
body.gutenberg-editor-page .editor-block-list__block[data-align="full"] {
    max-width: 1000px;
}
```

**Screenshots**
![_untitled____ _mindful_ _wordpress](https://user-images.githubusercontent.com/1231306/41612025-06b44f66-73c8-11e8-8fa8-d963eea6faf5.png)

**Comment by jasmussen** [comment]
So I was looking at this, and I'm wondering: is this solved by adding an editor style?

Right at this moment I'm leaning towards it being okay that the vanilla Gutenberg stylesheet assumes full width for full-wide. And given your theme does not, perhaps it's the domain of an editor style to mitigate that? There are instructions here: https://wordpress.org/gutenberg/handbook/extensibility/theme-support/#changing-the-width-of-the-editor — let me know if this isn't working or if there's nuance to this ticket that I'm missing upon further investigation.

**Comment by chrisvanpatten** [comment]
@jasmussen I think the problem boils down to the negative margins applied to a full-width block (`editor-block-list__block`) which offset the padding applied to the block list (`.editor-block-list__layout`), which is there to make side controls accessible.

Those negative margins make a truly full-width block stretch to the sides, but keep a container smaller than 100% (or more accurately, a container with no width or max-width, aka `auto`) from being centred on the page without also resetting those margins.

Let me know if all of that makes sense :)


Now all that said, I have a few thoughts:

1. Offsetting/undoing the negative margins might be best done in an editor stylesheet anyway, that's a totally valid point. I think it'd be nice if that wasn't necessary, but in the scheme of things this is a relatively tiny problem :)
2. It seems like the side UI is being de-emphasied as a whole with the More Menu changes. If the movers ever, well, move… we might find ourselves in a scenario where the side padding and negative offsets aren't really necessary anyway, which would be neat, and would make this issue null and void.

I'm going to close this for now as I think the solution (`margin-left: auto; margin-right: auto;` on a full-width block) is simple and easy. And there were other issues I was having with the positioning of the controls even when I undid those negative side margins, but those seem resolved (possibly in #9083) so this isn't as big a problem as before :)

**Comment by jasmussen** [comment]
This does make a lot of sense, and this is something to keep in mind as we continue to refine both editor style support as well as wide and fullwide support. It seems like a challenge that spans multiple individual aspects, and certainly worth solving. In one track, Riad is currently working on a runtime system to convert editor styles to base stylesheets that work in Gutenberg, that might take some of the vinegar out of this.
